### PR TITLE
Fix BaseRealm#checkIfValid() which does not throw Exception after close().

### PR DESCRIPTION
**Pull request by zaki50**
@realm/java review this please.

### CODE REVIEW

**Comment by cmelchior** [comment]
:+1:

File: realm/src/androidTest/java/io/realm/RealmTest.java [hunk 1]
```diff
@@ -186,6 +186,18 @@ public void testGetInstanceClearsCacheWhenFailed() {
...
 
+    public void testCheckValid() {
+        // checkIfValid() must not throw any Exception against valid Realm instance.
+        testRealm.checkIfValid();
+
+        testRealm.close();
+        try {
+            testRealm.checkIfValid();
+            fail("closed Realm instance must throw IllegalStateException.");
+        } catch (IllegalStateException e) {
```
Comment (beeender) [inline realm/src/androidTest/java/io/realm/RealmTest.java]: `e` -> `ignored`

**Comment by zaki50** [comment]
@beeender fixed

File: realm/src/androidTest/java/io/realm/RealmTest.java (hunk 1 above)
Comment (zaki50) [inline realm/src/androidTest/java/io/realm/RealmTest.java]: @beeender fixed

**Comment by beeender** [comment]
:+1:  Better squash and then push :smile:

**Pull request merged**